File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        unordered_map<char,int> mpp;
        mpp['b'] = 0;
        mpp['a'] = 0;
        mpp['l'] = 0;
        mpp['o'] = 0;
        mpp['n'] = 0;

        for(int i = 0 ; i <  text.size() ;i++){
            if(mpp.find(text[i]) != mpp.end()){
                mpp[text[i]]++;
            }
        }

        mpp['l'] = mpp['l']/2;
        mpp['o'] = mpp['o']/2;

        int minVal = INT_MAX ;

        for(auto it : mpp){
            minVal = min(it.second, minVal); 
        }

        return minVal ;



    }
};

File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.java
class Solution {
    public int maxNumberOfBalloons(String text) {

        HashMap<Character, Integer> mpp = new HashMap<>();

        mpp.put('b', 0);
        mpp.put('a', 0);
        mpp.put('l', 0);
        mpp.put('o', 0);
        mpp.put('n', 0);

        for (int i = 0; i < text.length(); i++) {

            char c = text.charAt(i);

            if (mpp.containsKey(c)) {
                mpp.put(c, mpp.get(c) + 1);
            }

        }

        mpp.put('l', mpp.get('l') / 2);
        mpp.put('o', mpp.get('o') / 2);

        int min = Integer.MAX_VALUE;

        for (Character c : mpp.keySet()) {
            if (c == 'b' || c == 'a' || c == 'n') {
                min = Math.min(min, mpp.get(c));
            }

        }

        int lVal = mpp.get('l');
        int oVal = mpp.get('o');

        min = Math.min(min, lVal);
        min = Math.min(min, oVal);

        return min;

    }
}